File: dashboards/cupid/backend/app/routers/sync.py
from fastapi import APIRouter, BackgroundTasks

from app.services import SyncService

router = APIRouter(prefix="/api/sync", tags=["sync"])
sync_service = SyncService()


@router.get("/status")
async def get_sync_status():
    """Get current sync status."""
    return await sync_service.get_status()


@router.post("/trigger")
async def trigger_sync(background_tasks: BackgroundTasks):
    """Manually trigger a sync."""
    background_tasks.add_task(sync_service.sync)
    return {"status": "started", "message": "Sync job started"}

File: dashboards/cupid/frontend/src/api/sync.ts
import { fetchApi } from './client'
import type { SyncStatus } from '@/types'

export async function getSyncStatus(): Promise<SyncStatus> {
  return fetchApi('/sync/status')
}

export async function triggerSync(): Promise<{ status: string; message: string }> {
  return fetchApi('/sync/trigger', { method: 'POST' })
}
